### DISCUSSION TITLE
Use the ARN created by 1 resource as a field in another resource of the same composition

### DISCUSSION THREAD

**Discussion by Sir-gits-a-lot** [crossplane/crossplane#4693] [Q&A]
I have use case where I create an IAM Role and then in another resource (EKS Addon) I have to mention the Role ARN.

How can I reference the ARN returned by status.atProvider.arn into a different resource of the same composition?

Is there a way to store the arn as a variable and then reference it in future?

If anyone has any idea please let me know.

**Comment by bobh66** [comment]
https://docs.crossplane.io/v1.13/concepts/composition/#patching-from-one-composed-resource-to-another-or-itself

There are example compositions here:  https://github.com/upbound/platform-ref-aws

**Reply by Sir-gits-a-lot** [reply]
Not from one Composed resource to another  but from one managed to another  managed resource within the same composition.

**Comment by bobh66** [comment]
A Composed resource is any resource in the list of resources for a Composition - it could be a Managed resource or another Composite resource.

The pattern is to patch from one Composed/Managed Resource to the Composite or the `environment` and then patch from the Composite or the `environment` into the other Composed/Managed Resource.

**Comment by Sir-gits-a-lot** [comment]
Thanks, it worked using CombineFromComposite.
